db/postgres.go

```diff
@@ -212,3 +212,36 @@ func NewPostgreSQL(uri, schema string) (PostgreSQL, error) {
 	}
 	return p, nil
 }
+
+func (p *PostgreSQL) ExtraIndexes(idxs []string) error {
+	for _, v := range idxs {
+		t := "((json->'" + v + "'))"
+		iname := "json_"
+		qsa := []string{"pais", "nome_socio", "codigo_pais", "faixa_etaria", "cnpj_cpf_do_socio", "qualificacao_socio", "codigo_faixa_etaria", "data_entrada_sociedade", "identificador_de_socio", "cpf_representante_legal", "nome_representante_legal", "codigo_qualificacao_socio", "qualificacao_representante_legal", "codigo_qualificacao_representante_legal"}
+		for _, v_qsa := range qsa {
+			if v == v_qsa {
+				t = fmt.Sprintf("(jsonb_extract_path(json, 'qsa', '%s') jsonb_ops)", v_qsa)
+				iname += "qsa_"
+				break
+			}
+		}
+		cnae := []string{"codigo", "descricao"}
+		for _, v_cnae := range cnae {
+			if v == v_cnae {
+				t = fmt.Sprintf("(jsonb_extract_path(json, 'cnae', '%s') jsonb_ops)", v_cnae)
+				iname += "cnae_"
+				break
+			}
+		}
+		q := fmt.Sprintf(
+			"CREATE INDEX IF NOT EXISTS idx_%s%s ON %s USING GIN %s;",
+			iname, v, p.CompanyTableName, t,
+		)
+		if _, err := p.pool.Exec(context.Background(), q); err != nil {
+			return fmt.Errorf("error to create indexe %s: %w", v, err)
+		}
+		log.Output(1, fmt.Sprintf("Indexes successfully created in the collection %s", p.CompanyTableName))
```

Comment:
Postgres não tem coleção, tem tabela ; )

```suggestion
		log.Output(1, fmt.Sprintf("Indexes successfully created for table %s", p.CompanyTableName))
```

Reply:
Copiar e colar com sono e cansado da nisso.... Não leio